sunkit_image/radial.py

```diff
@@ -577,3 +583,111 @@ def fnrgf(
         data[annulus] = ratio_mix[0] * smap.data[annulus] + ratio_mix[1] * data[annulus]
 
     return sunpy.map.Map(data, smap.meta)
+
+
+def rhef(
+    smap,
+    radial_bin_edges=None,
+    application_radius=0 * u.R_sun,
+    upsilon=0.35,
+    method="numpy",
+    vignette=None,
+):
+    """
+    Implementation of the Radial Histogram Equalizing Filter (RHEF).
+
+    The filter works as follows:
+
+    Radial Histogram Equalization is a simple algorithm for removing the radial gradient to reveal
+    coronal structure. It also significantly improves the visualization of high dynamic range solar imagery.
+    It takes the input map and bins the pixels by radius, then ranks each element sequentially and normalizes the pixels at that radius.
+
+    .. note::
+
+        After applying the filter, the maps need to be plotted with smap.plot(norm=False) to be visualized correctly.
+
+    Parameters
+    ----------
+    smap : `sunpy.map.Map`
+        The sunpy map to enhance.
+    radial_bin_edges : `astropy.units.Quantity`
+        A two-dimensional array of bin edges of size ``[2, nbins]`` where ``nbins`` is
+        the number of bins.
+    application_radius : `astropy.units.Quantity`, optional
+        The RHEF is applied to emission at radii above the application_radius.
+        Defaults to 0 solar radii.
+    upsilon : None or int or tuple of size 2, optional
+        A double-sided gamma function applied to the equalized histograms.
+        See Equation (4.15) in the thesis
+        Defaults to 0.35
+    method : str
+        A string describing which method to use for sorting.
+        Options are {"inplace", "numpy", "scipy"}.
+        Defaults to "numpy".
+    vignette : None or Bool or `astropy.units.Quantity`, optional
```

Comment:
I think this is too many options, it should be either an astropy Quantity or None (which can be default?)